### DISCUSSION TITLE
Specify User Agent

### DISCUSSION THREAD

**Discussion by therumbucket** [projectdiscovery/katana#270] [Ideas]
Is it possible to specify which specific user agent is sent out with katana's requests? I have tried using the -H parameter and adding the user-agent as a string, however, it appears it prepends the user-agent header with my info and contains the original header info.

**Comment by ehsandeep** [comment]
@therumbucket do you have an example where you noticed UA being prepended?

```console
echo  http://0.0.0.0:65161 | ./katana -H "User-Agent: test"
```

Results into:

```http
GET / HTTP/1.1
Host: 0.0.0.0:65161
Accept-Encoding: gzip
User-Agent: test
```

**Reply by heroesm** [reply]
Is it not possible to specify the user-agent in headless mode?
```
$ echo  http://0.0.0.0:65161 | ./katana -headless -H "User-Agent: test" -j

   __        __                
  / /_____ _/ /____ ____  ___ _
 /  '_/ _  / __/ _  / _ \/ _  /
/_/\_\\_,_/\__/\_,_/_//_/\_,_/							 

		projectdiscovery.io

[INF] Current katana version v1.0.1 (latest)
[INF] Started headless crawling for => http://0.0.0.0:65161
{"timestamp":"2023-04-20T17:58:21.140330753+08:00","request":{"method":"GET","endpoint":"http://0.0.0.0:65161","raw":"GET / HTTP/1.1\r\nHost: 0.0.0.0:65161\r\nUser-Agent: Go-http-client/1.1\r\nAccept-Encoding: gzip\r\n\r\n"},"error":"[hybrid:RUNTIME] navigation failed: net::ERR_CONNECTION_REFUSED \u003c- could not navigate target\n"}
```

In headless mode, the user-agent being used is not the specified one, nor is it similar to what a normal legitimate browser might use.

**Reply by ehsandeep** [reply]
@heroesm tracking this as new issue - https://github.com/projectdiscovery/katana/issues/414

**Comment by MetzinAround** [comment]
@therumbucket I'm gonna close this for now, but feel free to reopen if you'd like to.
